### DISCUSSION TITLE
Cannot Define config.ini Path

### DISCUSSION THREAD

**Discussion by Brettem** [UserBlackBox/octoprint-cli#29] [Q&A]
Describe the bug:
I am unable to get octoprint-cli to recognize it's own config file. I created a new directory on my local device and created a `config.ini` file within it using the template provided. However, no matter how I try to call the `-c` command I cannot define the config path so no other function will run.

To Reproduce

```
pip install octoprint-cli

octoprint-cli -c .\config.ini
>>>Invalid or Missing Arguments

octoprint-cli -c ".\config.ini"
>>>Invalid or Missing Arguments

octoprint-cli -c=.\config.ini
>>>usage: octoprint-cli [-h] [-c CONFIG_PATH] [-v]
                     {version,continuous,layers,gcode,print,connection,temp,system,files}
                     ...
>>>octoprint-cli: error: invalid choice: '.\\config.ini' (choose from 'version', 'continuous', 'layers', 'gcode', 'print', 'connection', 'temp', 'system', 'file
s')

octoprint-cli version
>>>Configuration file is not complete or does not exist
```

No matter what I try I am unable to get the app to recognize the config file I created. It appears that the octoprint-cli doesn't like my argument format but I'm not sure how else to configure it if I am just trying to define the path to the config file required for all positional arguments. 

Expected behavior:
After `cd` into directory where created config file is located, run `>octoprint-cli -c ".\config.ini"` and config path is redefined.

Environment Information:

OS: [e.g. iOS] both OctoPi and Windows 11
Version 3.3.2 (cuurent)
Python Version: 3,8,8
Dependency versions
OctoPrint version: current
3D Printer Model: Prusa i3 Mk3S+

**Comment by hexadecimalDinosaur** [comment]
can you try `octoprint-cli -c config.ini`

**Reply by Brettem** [reply]
Wow thank you for the quick response. 
When I run `octoprint-cli -c config.ini` It still just gives me `Invalid or Missing Arguments`

**Reply by hexadecimalDinosaur** [reply]
Will probably have to investigate further on a Windows system, I'm not seeing this issue on Linux at the moment
#30

**Reply by Brettem** [reply]
I did a bit of digging into the `_main_.py` file for octoprint-cli and found that the specific line which is calling `Configuration file is not complete or does not exist` when I run a command like `octoprint-cli version` is from line 858 and further the result of `if loadConfig(os.path.join(sys.path[0], 'config.ini'))` failing.

So when I looked into what was being fed into loadConfig I found that `os.path.join(sys.path[0], 'config.ini')` is being interpreted as `C:\Users\[MY_USER]\anaconda3\Scripts\octoprint-cli.exe\config.ini` is this the intended path for default config?

**Reply by Brettem** [reply]
I decided to try and move the config file into the octoprint_cli package directory and I was also able to confirm that when I run `octoprint-cli -c C:\Users\[MY_USER]\anaconda3\Lib\site-packages\octoprint_cli\config.ini` octoprint_cli does recognize `options.config_path` as `C:\Users\brett\anaconda3\Lib\site-packages\octoprint_cli\config.ini` but still calls `Invalid or Missing Arguments` so if I am interpreting this correctly this may be an issue with the parser instead of the path argument itself.

**Reply by Brettem** [reply]
It appears that the error being called that results in the `Invalid or Missing Arguments` statement is `AttributeError: 'Namespace' object has no attribute 'func'` because of `options.func(options)`
